src/KeycloakContext.ts

```diff
@@ -1,21 +1,20 @@
 import { AuthContextProvider } from './api'
+import Keycloak from 'keycloak-connect'
 
 export class KeycloakContext implements AuthContextProvider {
-  public readonly request: any
-  public readonly accessToken: any
-  public readonly authenticated: boolean
+  public readonly request: Keycloak.GrantedRequest
+  public readonly accessToken: Keycloak.Token | undefined
 
-  constructor ({ req }: { req: any }) {
+  constructor ({ req }: { req: Keycloak.GrantedRequest }) {
     this.request = req
     this.accessToken = (req && req.kauth && req.kauth.grant) ? req.kauth.grant.access_token : undefined
-    this.authenticated = this.accessToken && !this.accessToken.isExpired()
   }
 
   public isAuthenticated (): boolean {
-    return this.authenticated
+    return (this.accessToken && !this.accessToken.isExpired()) ? true : false 
   }
 
   public hasRole (role: string): boolean {
-    return this.isAuthenticated() && this.accessToken.hasRole(role)
+    return ((this.accessToken && !this.accessToken.isExpired()) && this.accessToken.hasRole(role)) ? true : false
```

Comment:
```suggestion
    return this.isAuthenticated() && this.accessToken.hasRole(role)
```

Reply:
Again, similar to the other comment. If we use that, the compiler says something like `this.accessToken` is possibly undefined. The compiler is not smart enough to see that `this.isAuthenticated()` is actually doing that check.

Reply:
In this cases, I'm simply disabling compiler for the line :)

`// @ts-ignore`

Reply:
By the end of the day you want to have great js code. 
If your code goes around typescript limitations then we can disable it.

Reply:
Thanks for that tip 😃